prefix_db.go

```diff
@@ -293,29 +331,45 @@ func (itr *prefixIterator) Valid() bool {
 	return itr.valid && itr.source.Valid()
 }
 
-func (itr *prefixIterator) Next() {
+func (itr *prefixIterator) Next() error {
 	if !itr.valid {
-		panic("prefixIterator invalid, cannot call Next()")
+		return errors.New("prefixIterator invalid; cannot call Next()")
 	}
-	itr.source.Next()
-	if !itr.source.Valid() || !bytes.HasPrefix(itr.source.Key(), itr.prefix) {
+	err := itr.source.Next()
+	if err != nil {
+		return err
+	}
+	key, err := itr.source.Key()
+	if err != nil {
+		return err
+	}
+	if !itr.source.Valid() || !bytes.HasPrefix(key, itr.prefix) {
 		itr.valid = false
-		return
+		return nil
 	}
+	return nil
 }
 
-func (itr *prefixIterator) Key() (key []byte) {
+func (itr *prefixIterator) Key() (key []byte, err error) {
 	if !itr.valid {
-		panic("prefixIterator invalid, cannot call Key()")
```

Comment:
panic is appropriate here as it checks program logic (ie. nobody should call Key/Value if iterator is not Valid)